parsec/core/local_db.py

```diff
@@ -26,68 +31,339 @@ def __init__(self, access):
+
+    # Generic manifest operations
+
+    def _check_presence(self, conn: Connection, access: Access):
+        cursor = conn.cursor()
+        res = cursor.execute(
+            "SELECT manifest_id FROM manifests WHERE manifest_id = ?", (str(access.id),)
+        )
+        result = res.fetchone()
+        cursor.close()
+        return bool(result)
 
-    def get(self, access: Access):
-        file = self._find(access)
+    def _get_manifest(self, conn: Connection, access: Access):
+        cursor = conn.cursor()
+        res = cursor.execute(
+            "SELECT manifest_id, blob FROM manifests WHERE manifest_id = ?", (str(access.id),)
+        )
+        manifest = res.fetchone()
+        cursor.close()
+        if not manifest:
+            raise LocalDBMissingEntry(access)
+        return decrypt_raw_with_secret_key(access.key, manifest[1])
+
+    def _set_manifest(self, conn: Connection, access: Access, raw: bytes):
+        assert isinstance(raw, (bytes, bytearray))
+        ciphered = encrypt_raw_with_secret_key(access.key, raw)
+
+        cursor = conn.cursor()
+        cursor.execute("DELETE FROM manifests WHERE manifest_id = ? ", (str(access.id),))
+        cursor.execute(
+            """INSERT INTO manifests (manifest_id, blob)
+               VALUES (?, ?)""",
+            (str(access.id), ciphered),
+        )
+        cursor.close()
+
+    def _clear_manifest(self, conn: Connection, access: Access):
+        cursor = conn.cursor()
+        cursor.execute("DELETE FROM manifests WHERE manifest_id = ?", (str(access.id),))
+        cursor.execute("SELECT changes()")
+        deleted, = cursor.fetchone()
+        cursor.close()
+        if not deleted:
+            raise LocalDBMissingEntry(access)
+
+    # Remote manifest operations
+
+    def get_remote_manifest(self, access: Access):
+        return self._get_manifest(self.remote_conn, access)
+
+    def set_remote_manifest(self, access: Access, raw: bytes):
+        if self._check_presence(self.local_conn, access):
+            raise ValueError("Cannot set remote manifest: a local manifest is already present")
+        self._set_manifest(self.remote_conn, access, raw)
+
+    def clear_remote_manifest(self, access: Access):
+        self._clear_manifest(self.remote_conn, access)
+
+    # Local manifest operations
+
+    def get_local_manifest(self, access: Access):
+        return self._get_manifest(self.local_conn, access)
+
+    def set_local_manifest(self, access: Access, raw: bytes):
+        if self._check_presence(self.remote_conn, access):
+            raise ValueError("Cannot set local manifest: a remote manifest is already present")
+        self._set_manifest(self.local_conn, access, raw)
+
+    def clear_local_manifest(self, access: Access):
+        self._clear_manifest(self.local_conn, access)
+
+    # Generic block operations
+
+    def _get_block(self, conn: Connection, path: Path, access: Access):
+        file = path / str(access.id)
         if not file:
```

Comment:
T'es sûr que ça marche ça ? j'aurai plutôt fait `if not file.exists()`

Reply:
Non ca marche pas du tout --' 

Inquiétant de voir que les tests n'ont pas détecté le problème, je vais investiguer ca.